pkg/controller/operators/catalog/operator.go

```diff
@@ -1857,11 +1857,15 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 					return errorwrap.Wrapf(err, "error decoding %s object to an unstructured object", step.Resource.Name)
 				}
 
-				// Get the resource from the GVK.
+				// Get the GVK from the resource.
+				// Then check if the API for this dynamic resource is available on-cluster, before creating the resource, by querying discovery.
+				// This ensures CRs created via the dynamic client are created after the corresponding CRD is accepted and established on the cluster to avoid potential race conditions.
 				gvk := unstructuredObject.GroupVersionKind()
 				r, err := o.apiresourceFromGVK(gvk)
 				if err != nil {
-					return err
+					o.logger.Infof("failed to find any server resources for GVK %s in api discovery", gvk.String())
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusUnknown
```

Comment:
I'm not even sure this is necessary - isn't the step in an `Unknown` state at this point anyway? Maybe just the `continue` is enough